src/PowerShellEditorServices/Services/Symbols/SymbolsService.cs

```diff
@@ -408,123 +299,56 @@ static string[] GetIdentifiers(string symbolName, SymbolType symbolType, Command
-                    foundSymbol.ScriptRegion,
-                    foundSymbol.FilePath,
-                    foundSymbol.SourceLine);
-            }
-
-            ScriptFile[] referencedFiles =
-                _workspaceService.ExpandScriptReferences(
-                    sourceFile);
-
-            HashSet<string> filesSearched = new(StringComparer.OrdinalIgnoreCase);
-
-            // look through the referenced files until definition is found
-            // or there are no more file to look through
-            SymbolReference foundDefinition = null;
-            foreach (ScriptFile scriptFile in referencedFiles)
-            {
-                foundDefinition = AstOperations.FindDefinitionOfSymbol(scriptFile.ScriptAst, foundSymbol);
-
-                filesSearched.Add(scriptFile.FilePath);
-                if (foundDefinition is not null)
+                foreach (SymbolReference foundReference in symbols)
                 {
-                    foundDefinition.FilePath = scriptFile.FilePath;
-                    break;
-                }
-
-                if (foundSymbol.SymbolType == SymbolType.Function)
-                {
-                    // Dot-sourcing is parsed as a "Function" Symbol.
-                    string dotSourcedPath = GetDotSourcedPath(foundSymbol, scriptFile);
-                    if (scriptFile.FilePath == dotSourcedPath)
+                    if (foundReference.IsDeclaration)
                     {
-                        foundDefinition = new SymbolReference(SymbolType.Function, foundSymbol.SymbolName, scriptFile.ScriptAst.Extent, scriptFile.FilePath);
-                        break;
+                        _logger.LogDebug($"Found possible declaration in same file ${foundReference}");
+                        declarations.Add(foundReference);
                     }
                 }
             }
 
-            // if the definition the not found in referenced files
-            // look for it in all the files in the workspace
-            if (foundDefinition is null)
+            if (declarations.Any())
             {
-                // Get a list of all powershell files in the workspace path
-                foreach (string file in _workspaceService.EnumeratePSFiles())
-                {
-                    if (filesSearched.Contains(file))
-                    {
-                        continue;
-                    }
-
-                    foundDefinition =
-                        AstOperations.FindDefinitionOfSymbol(
-                            Parser.ParseFile(file, out Token[] tokens, out ParseError[] parseErrors),
-                            foundSymbol);
-
-                    filesSearched.Add(file);
-                    if (foundDefinition is not null)
-                    {
-                        foundDefinition.FilePath = file;
-                        break;
-                    }
-                }
+                return declarations;
```

Comment:
```suggestion
                return declarations.Where((d) =>
                    symbol.SymbolType == d.SymbolType ||
                    (symbol.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type
                        && d.SymbolType is SymbolType.Class or SymbolType.Enum) ||
                    (symbol.SymbolType is SymbolType.EnumMember or SymbolType.Property
                        && d.SymbolType is SymbolType.EnumMember)
                );
```

Fix go to definition symbol type mismatch (part 1)